source/data-encryption.txt

```diff
@@ -12,59 +12,167 @@ Data Encryption
+only decrypted on the client side. 
+
+You can selectively encrypt individual fields within a document, 
+multiple fields within the document, or the entire document. You can optionally
+secure each field with its own key and decrypt them seamlessly 
+on the client by using a MongoDB driver. CSFLE uses AES-256 in 
+authenticated CBC mode to encrypt data.
+
+The following diagram demonstrates a CSFLE workflow where user records 
+are stored in a MongoDB database and queried by the client. The user's 
+social security number (SSN) is encrypted before being stored in the 
+database. When the application submits a basic equality query on the field, 
+the MongoDB driver uses the key to encrypt the query and decrypt the query results, 
+before returning them to the authenticated client as readable plaintext.
+
+.. figure:: /includes/images/csfle-encryption.svg
+   :figwidth: 750px
+   :alt: An image showing an example client-side field-level encryption (CSFLE) workflow.
 
 Queryable Encryption
-####################
-
-:ref:`Queryable Encryption <qe-manual-feature-qe>` helps organizations protect sensitive data when it is queried.
-It allows applications to encrypt sensitive data on the client side, securely store it in the database, and perform equality
-and range queries directly on the encrypted data.
-This ensures protection for sensitive information without sacrificing the ability to perform queries on it.
+````````````````````
+
+:ref:`Queryable Encryption <qe-manual-feature-qe>` helps organizations 
+protect sensitive data when it is queried. Like CSFLE, it allows applications 
+to encrypt your data on the client side before storing it in the MongoDB database. 
+It also enables applications to perform expressive queries directly on the 
+encrypted data by using an encrypted search algorithm. This ensures protection 
+for sensitive information without sacrificing the ability to perform queries on it.
+
+Queryable encryption uses AES-256 in authenticated CBC mode
+to encrypt data, and uses a combination of AES-256, SHA2, and HMACs
+for its encrypted search algorithm.
+
+The following diagram demonstrates a queryable encryption workflow where 
+user records are stored in a MongoDB database and queried by the client. 
+The user's date of birth (DOB) is encrypted before being stored in the 
+database. When the application submits an expressive range query on the field, 
+the MongoDB driver uses the key to encrypt the query and passes a cryptographic token 
+with it to the MongoDB server. The server uses the encrypted search algorithm 
+to process the query without knowing the actual data. Finally, the driver uses the 
+key to decrypt the query results and returns them to the authenticated 
+client as readable plaintext.
+
+.. figure:: /includes/images/queryable-encryption.svg
+   :figwidth: 750px
+   :alt: An image showing an example queryable encryption workflow.
 
 Recommendations
-~~~~~~~~~~~~~~~
+---------------
```

Comment:
Can we change this to "Recommendations for Data Encryption"? See above comment about reducing repeated headings across a docs set